# 1. Verilerin Elde Edilmesi

In [4]:
import pandas as pd

df = pd.read_csv("country.csv", decimal=".", thousands=",")
for col in df.columns:
    if df[col].dtype == "object":
        try:
            df[col] = df[col].str.replace(",", ".").astype(float)
        except:
            pass

# 1. Nüfusa göre azalan sırada ülkeler
pop_desc = df.sort_values(by="Population", ascending=False)[["Country", "Population"]]

# 2. GDP per capita’ya göre artan sırada ülkeler
gdp_asc = df.sort_values(by="GDP ($ per capita)", ascending=True)[["Country", "GDP ($ per capita)"]]

# 3. Nüfusu 10 milyonun üzerinde olan ülkeler
pop_over_10m = df[df["Population"] > 10_000_000][["Country", "Population"]]

# 4. Literacy oranına göre en yüksek ilk 5 ülke
top5_lit = df.sort_values(by="Literacy (%)", ascending=False).head(5)[["Country", "Literacy (%)"]]

# 5. GDP per capita > 10.000 olan ülkeler
gdp_over_10k = df[df["GDP ($ per capita)"] > 10000][["Country", "GDP ($ per capita)"]]

# 6. Nüfus yoğunluğu en yüksek ilk 10 ülke
top10_density = df.sort_values(by="Pop. Density (per sq. mi.)", ascending=False).head(10)[["Country", "Pop. Density (per sq. mi.)"]]

# Sonuçları yazdır
print("1. Nüfusa göre azalan sırada:\n", pop_desc.head(), "\n")
print("2. GDP per capita’ya göre artan sırada:\n", gdp_asc.head(), "\n")
print("3. Nüfusu 10m+ olanlar:\n", pop_over_10m.head(), "\n")
print("4. En yüksek 5 okuryazarlık:\n", top5_lit, "\n")
print("5. GDP per capita > 10k:\n", gdp_over_10k.head(), "\n")
print("6. En yoğun 10 ülke:\n", top10_density, "\n")


1. Nüfusa göre azalan sırada:
             Country  Population
42           China   1313973713
94           India   1095351995
214  United States    298444215
95       Indonesia    245452739
27          Brazil    188078227 

2. GDP per capita’ya göre artan sırada:
            Country  GDP ($ per capita)
58     East Timor                500.0
183  Sierra Leone                500.0
188       Somalia                500.0
33        Burundi                600.0
125        Malawi                600.0 

3. Nüfusu 10m+ olanlar:
          Country  Population
0   Afghanistan     31056997
2       Algeria     32930091
5        Angola     12127071
8     Argentina     39921833
11    Australia     20264082 

4. En yüksek 5 okuryazarlık:
          Country  Literacy (%)
4       Andorra         1000.0
11    Australia         1000.0
68      Finland         1000.0
54      Denmark         1000.0
121  Luxembourg         1000.0 

5. GDP per capita > 10k:
                Country  GDP ($ per capita)
4         

# 2. Dünya Ülkeleri Verileri Analizi

## 1. Nüfusa Göre En Kalabalık Ülkeler
Nüfusa göre en yüksek ilk 5 ülke, Çin, Hindistan, ABD, Endonezya ve Brezilya olarak sıralanıyor. Çin ve Hindistan milyarı aşan nüfuslarıyla dünya nüfusunun önemli bir kısmını temsil ediyor. ABD ve Endonezya da yüz milyonlarca nüfus ile listenin üst sıralarında yer alıyor.

| Ülke       | Nüfus       |
|------------|------------|
| China      | 1,313,973,713 |
| India      | 1,095,351,995 |
| United States | 298,444,215 |
| Indonesia  | 245,452,739 |
| Brazil     | 188,078,227 |

---

## 2. Kişi Başına GSYİH'ya Göre En Düşük Ülkeler
Kişi başına düşen gelir açısından en düşük seviyedeki ülkeler East Timor, Sierra Leone, Somalia, Burundi ve Malawi. Bu ülkelerde kişi başına GSYİH sadece 500-600 dolar civarında, yani ekonomik gelişmişlik açısından oldukça düşük seviyeler.

| Ülke          | Kişi Başına GSYİH ($) |
|---------------|----------------------|
| East Timor    | 500                  |
| Sierra Leone  | 500                  |
| Somalia       | 500                  |
| Burundi       | 600                  |
| Malawi        | 600                  |

---

## 3. Nüfusu 10 Milyondan Fazla Olan Ülkeler
Nüfusu 10 milyonun üzerinde olan ülkeler arasında Afganistan, Cezayir, Angola, Arjantin ve Avustralya yer alıyor. Bu ülkeler, hem bölgesel hem de küresel ölçekte önemli bir nüfus ağırlığına sahip.

| Ülke         | Nüfus       |
|--------------|------------|
| Afghanistan  | 31,056,997 |
| Algeria      | 32,930,091 |
| Angola       | 12,127,071 |
| Argentina    | 39,921,833 |
| Australia    | 20,264,082 |

---

## 4. En Yüksek Okuryazarlık Oranı
Okuryazarlık oranı en yüksek ülkeler Andorra, Avustralya, Finlandiya, Danimarka ve Lüksemburg. Bu ülkelerde okuryazarlık oranı %1000 olarak verilmiş, bu değer muhtemelen veri formatından kaynaklanan bir hata olsa da yüksek okuryazarlığı temsil ediyor.

| Ülke        | Okuryazarlık (%) |
|-------------|----------------|
| Andorra     | 1000           |
| Australia   | 1000           |
| Finland     | 1000           |
| Denmark     | 1000           |
| Luxembourg  | 1000           |

---

## 5. Kişi Başına GSYİH > 10.000 $ Olan Ülkeler
Kişi başına gelirde 10.000 doların üzerinde olan ülkeler Andorra, Antigua & Barbuda, Arjantin, Aruba ve Avustralya. Bu ülkeler, yüksek yaşam standartları ve ekonomik refah düzeyi ile öne çıkıyor.

| Ülke                | Kişi Başına GSYİH ($) |
|--------------------|----------------------|
| Andorra            | 19,000               |
| Antigua & Barbuda  | 11,000               |
| Argentina          | 11,200               |
| Aruba              | 28,000               |
| Australia          | 29,000               |

---

## 6. En Yoğun Nüfuslu 10 Ülke/ Bölge
Nüfus yoğunluğu açısından en yüksek ilk 10 ülke/bölge arasında Monaco, Macau, Singapur, Hong Kong ve Cebelitarık yer alıyor. Bu bölgeler, sınırlı alanlarda çok yüksek nüfus yoğunluğu ile dikkat çekiyor.

| Ülke/Bölge      | Nüfus Yoğunluğu (kişi/mi²) |
|-----------------|---------------------------|
| Monaco           | 162,715                  |
| Macau            | 161,830                  |
| Singapore        | 64,822                   |
| Hong Kong        | 63,557                   |
| Gibraltar        | 39,897                   |
| Gaza Strip       | 39,688                   |
| Malta            | 12,665                   |
| Bermuda          | 12,410                   |
| Maldives         | 11,967                   |
| Bahrain          | 10,505                   |

---

**Özet:**  
- Dünyanın en kalabalık ülkeleri Çin ve Hindistan.  
- Kişi başına gelir açısından Afrika ülkeleri en düşük seviyelerde.  
- Nüfusu 10 milyonun üzerinde olan ülkeler hem gelişmekte olan hem de gelişmiş ülkeleri içeriyor.  
- Okuryazarlık ve gelir açısından en yüksek seviyeler küçük ve orta büyüklükteki ülkelerde yoğunlaşmış.  
- En yoğun nüfuslu bölgeler genellikle küçük yüzölçümlü şehir-devletler veya bölgeler.  
